api/src/middleware/__tests__/assessmentsRouter.ts

```diff
@@ -59,17 +81,520 @@ describe('assessmentsRouter', () => {
+        .expect(
+          404,
+          errorEnvelope('The requested resource does not exist.'),
+          err => {
+            expect(mockGetProgramIdByProgramAssessmentId).toHaveBeenCalledWith(
+              programAssessmentId
+            );
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with a bad request error if given an invalid submission id ', done => {
+      const programAssessmentId = 1,
+        submissionId = 'test';
+
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(400, done);
+    });
+
+    it('should respond with a NotFoundError if the submission ID was not found in the database ', done => {
+      const programAssessmentId = 1,
+        submissionId = 8;
+
+      mockSubmissionDetails.mockResolvedValue([]);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(
+          404,
+          errorEnvelope('The requested resource does not exist.'),
+          err => {
+            expect(mockSubmissionDetails).toHaveBeenCalledWith(
+              programAssessmentId,
+              submissionId,
+              true
+            );
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with an UnauthorizedError if the logged-in principal ID is not the same as the principal ID of the submission ID and is not the principal ID of the program facilitator', done => {
+      const loggedPrincipalId = 4;
+      const participantPrincipalId = 2;
+      const programAssessmentId = 1;
+      const submissionId = 1;
+
+      const assessmentSubmission = {
+        id: 2,
+        assessment_id: programAssessmentId,
+        principal_id: participantPrincipalId,
+        assessment_submission_state: 'Submitted',
+        opened_at: '2023-02-09 12:00:00',
+        submitted_at: '2023-02-09 13:23:45',
+        responses: [
+          {
+            id: 1,
+            answer_id: 1,
+            assessment_id: 1,
+            submission_id: 2,
+            question_id: 1,
+          },
+        ],
+      };
+
+      mockSubmissionDetails.mockResolvedValue([assessmentSubmission]);
+
+      mockPrincipalId(loggedPrincipalId);
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(401, errorEnvelope('Unauthorized user.'), err => {
+          expect(mockSubmissionDetails).toHaveBeenCalledWith(
+            programAssessmentId,
+            submissionId,
+            true
+          );
+          done(err);
+        });
+    });
+
+    it('should respond with an internal server error if a database error occurs', done => {
+      const programAssessmentId = 1,
+        submissionId = 1;
+      mockSubmissionDetails.mockRejectedValue(new Error());
+      appAgent;
+
+      appAgent
+        .get(`/${programAssessmentId}/submissions/${submissionId}`)
+        .expect(500, done);
```

Comment:
You'll want a error envelope here for sure, since we need to specify that we would be calling our service file functions.